### PR TITLE
Dhcp Relay Global

### PR DESCRIPTION
**Pull request by saichint**
This PR is for adding a new provider cisco_dhcp_relay_global. All tests pass on all nxos platforms.
Updated README file. Noticed that cisco_bfd_global entry was missing in the support matrix, added that as well.

### CODE REVIEW

File: lib/puppet/provider/cisco_dhcp_relay_global/cisco.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      else
+        current_state[prop] = val ? :true : :false
+      end
+    end
+    new(current_state)
+  end # self.properties_get
+
+  def self.instances
+    globals = []
+    Cisco::DhcpRelayGlobal.globals.each do |global_id, nu_obj|
```
Comment (mikewiebe) [inline lib/puppet/provider/cisco_dhcp_relay_global/cisco.rb]: So just realized that `DhcpRelayGlobal.globals` enables the feature.  Not sure we want to do this, but let's discuss this more when you get in.

**Comment by mikewiebe** [comment]
Code looks good.  Just have one area I want to discuss with you.

File: README.md [hunk 2]
```diff
@@ -1635,6 +1641,78 @@ The bridge-domain ID. Valid values are one or range of integers.
 The Virtual Network Identifier (VNI) id that is mapped to the VLAN. Valid values are one or range of integers
 
 --
+### Type: cisco_dhcp_relay_global
+
+Manages configuration of a DHCP relay global configuration.
+
+| Platform | OS Minimum Version | Module Minimum Version |
+|----------|:------------------:|:----------------------:|
+| N9k      | 7.0(3)I3(1)        | 1.4.0                  |
```
Comment (mikewiebe) [inline README.md]: Based on the test results, the `OS Minimum Version` should be `7.0.3.I2.2e` right?

Comment (mikewiebe) [inline README.md]:
```
727: 2016-09-14T17:04:41: %LIB-6-INFO: Node under test:
 728: 2016-09-14T17:04:41: %LIB-6-INFO:   - name  - dt-n9k5-1
 729: 2016-09-14T17:04:41: %LIB-6-INFO:   - type  - N9K-C9396PX
 730: 2016-09-14T17:04:41: %LIB-6-INFO:   - image - bootflash:///nxos.7.0.3.I2.2e.bin
 731: 2016-09-14T17:04:41: %LIB-6-INFO: 
 732: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_information_option_vpn = 5.59 s = .
 733: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_smart_relay = 3.90 s = .
 734: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_information_option = 5.73 s = .
 735: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_information_trust_all = 5.63 s = .
 736: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_src_intf = 5.65 s = .
 737: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv6_option_vpn = 5.58 s = .
 738: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_sub_option_circuit_id_custom = 3.77 s = S
 739: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv6_src_intf = 4.02 s = .
 740: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv6_option_cisco = 5.63 s = .
 741: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_destroy = 5.76 s = .
 742: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_relay = 5.77 s = .
 743: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_sub_option_circuit_id_string = 5.18 s = .
 744: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_src_addr_hsrp = 5.22 s = .
 745: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv6_relay = 5.76 s = .
 746: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_sub_option_cisco = 5.59 s = .
 747: 2016-09-14T17:04:41: %LIB-6-INFO: TestDhcpRelayGlobal#test_ipv4_information_option_trust = 7.17 s = .
 748: 2016-09-14T17:04:41: %LIB-6-INFO: 
 749: 2016-09-14T17:04:41: %LIB-6-INFO: Finished in 87.165507s, 0.1836 runs/s, 0.5163 assertions/s.
 750: 2016-09-14T17:04:41: %LIB-6-INFO: 
 751: 2016-09-14T17:04:41: %LIB-6-INFO:   1) Skipped:
 752: 2016-09-14T17:04:41: %LIB-6-INFO:
 753: 2016-09-14T17:04:41: %LIB-6-INFO: This property is not supported on Nexus 'I2' images
 754: 2016-09-14T17:04:41: %LIB-6-INFO: 
 755: 2016-09-14T17:04:41: %LIB-6-INFO: 16 runs, 45 assertions, 0 failures, 0 errors, 1 skips
```

File: lib/puppet/provider/cisco_dhcp_relay_global/cisco.rb (hunk 1 above)
Comment (saichint) [inline lib/puppet/provider/cisco_dhcp_relay_global/cisco.rb]: Will open a PR on node utils for this.

**Pull request merged**